path/_util.ts

```diff
@@ -131,3 +132,63 @@ export function encodeWhitespace(string: string): string {
     return WHITESPACE_ENCODINGS[c] ?? c;
   });
 }
+
+export function lastPathSegment(
+  path: string,
+  isSep: (char: number) => boolean,
+  start = 0,
+): string {
+  let matchedNonSeparator = false;
+  let end = path.length;
+
+  for (let i = path.length - 1; i >= 0; --i) {
```

Comment:
`i >= 0` should be `i >= start`.

Reply:
Good catch. It also exposed one bug, which I also fixed.